lib/medusa/consumer/rabbitmq.ex

```diff
@@ -60,4 +70,34 @@ defmodule Medusa.Consumer.RabbitMQ do
     end
   end
 
+  defp do_event(%{metadata: metadata} = message, function) do
+    try do
+      case function.(message) do
+        :error ->
+          retry_event(message)
+        {:error, _} ->
+          retry_event(message)
+        _ ->
+          AMQP.Basic.ack(metadata["channel"], metadata["delivery_tag"])
+      end
+    rescue
+      _ -> retry_event(message)
+    catch
+      _ -> retry_event(message)
+    end
+  end
+
+  defp retry_event(%Message{} = message) do
+    message = update_in(message,
+                        [Access.key(:metadata), "retry"],
+                        &((&1 || 0) + 1))
+    time =
+      Medusa.config
+      |> Keyword.get(:retry_consumer_pow, 2)
```

Comment:
this shouldn't be configurable, can be constant. btw it's a "base" instead of "power"